pkg/server/delta/v3/server.go

```diff
@@ -162,25 +162,25 @@ func (s *server) processDelta(str stream.DeltaStream, reqCh <-chan *discovery.De
 				// Since there was no previous state, we know we're handling the first request of this type
 				// so we set the initial resource versions if we have any, and also signal if this stream is in wildcard mode.
 				watch.state = stream.NewStreamState(len(req.GetResourceNamesSubscribe()) == 0, req.GetInitialResourceVersions())
-				watch.responses = make(chan cache.DeltaResponse, 1)
-
+			} else {
 				// We must signal goroutine termination to prevent a race between the cancel closing the watch
```

Comment:
I think we can actually do away with this comment as well. Or maybe note that we now close the responses channel when canceling an old watch.

Reply:
I deleted the comment.